Classes/Library/Authentication.php

```diff
@@ -686,13 +686,45 @@ public static function merge(array $ldap = [], array $typo3 = [], array $mapping
             $backupTSFE = $GLOBALS['TSFE'];
 
             // Advanced stdWrap methods require a valid $GLOBALS['TSFE'] => create the most lightweight one
-            $GLOBALS['TSFE'] = GeneralUtility::makeInstance(
-                \TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController::class,
-                $GLOBALS['TYPO3_CONF_VARS'],
-                0,
-                ''
-            );
-            $GLOBALS['TSFE']->initTemplate();
+            $typo3Branch = class_exists(\TYPO3\CMS\Core\Information\Typo3Version::class)
+                ? (new \TYPO3\CMS\Core\Information\Typo3Version())->getBranch()
+                : TYPO3_branch;
+            if (version_compare($typo3Branch, '10.0', '<')) {
+                // Keep the old approach to init TSFE before Typo3 v10
+                $GLOBALS['TSFE'] = GeneralUtility::makeInstance(
+                    \TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController::class,
+                    $GLOBALS['TYPO3_CONF_VARS'],
+                    0,
+                    ''
+                );
+                $GLOBALS['TSFE']->initTemplate();
+            } else {
+                // Use a modern approach to init TSFE on TYPO3 v10+
+                // Use SiteFinder to get a Site object for the current page tree
+                $siteFinder = GeneralUtility::makeInstance(
+                    \TYPO3\CMS\Core\Site\SiteFinder::class
+                );
+                $currentSite = $siteFinder->getSiteByPageId($typo3['uid']);
```

Comment:
What is `$typo3['uid']`? Is this the correct value to use here? 

E.g.: An exception is thrown if `$typo3['uid']` points to a PID that doesn't exist (anymore).

Reply:
Maybe this would be a viable solution: 

```
$allSites = $siteFinder->getAllSites();
$currentSite = reset($allSites);
```

Reply:
Just in case somebody finds this in the future (i was very busy, sorry):
I mixed up the variables. I thought `$typo3` would refer to something like the TSFE, but it just contains the feuser. What i actually needed at that point was the UID of the page.

I managed to get it working better with `$typo3['pid']` here, as this would at least refer to the correct site in a multisite installation if the feuser is located somewhere in the correct tree.

The `reset($siteFinder->getAllSites())` suggestion might also be quick fix, but i originally dismissed that because it might get the wrong site in a multisite installation and introduce subtle errors down the line.